locality_alignment/backbones/dbot_random.py

```diff
@@ -0,0 +1,211 @@
+# Modified from https://github.com/liuxingbin/dbot
+
+import os
+import math
+import logging
+from functools import partial
+import torch
+import torch.nn as nn
+
+import timm
+from timm.models.vision_transformer import VisionTransformer
+from timm.models.registry import register_model
+
+
+class dBOTRandomVisionTransformer(VisionTransformer):
+    """Vision Transformer with support for global average pooling."""
+
+    def __init__(self, global_pool=False, **kwargs):
+        super().__init__(**kwargs)
+
+        self.global_pool = global_pool
+        if self.global_pool:
+            norm_layer = kwargs["norm_layer"]
+            embed_dim = kwargs["embed_dim"]
+            self.fc_norm = norm_layer(embed_dim)
+
+            del self.norm  # remove the original norm
+
+    def forward_features(self, x):
+        B = x.shape[0]
+        x = self.patch_embed(x)
+
+        # stole cls_tokens impl from Phil Wang, thanks
+        cls_tokens = self.cls_token.expand(B, -1, -1)
+        x = torch.cat((cls_tokens, x), dim=1)
+        x = x + self.pos_embed
+        x = self.pos_drop(x)
+
+        for blk in self.blocks:
+            x = blk(x)
+
+        # global pool or cls token
+        if self.global_pool:
+            x = x[:, 1:, :].mean(dim=1)
+            outcome = self.fc_norm(x)
+        else:
+            x = self.norm(x)
+            outcome = x[:, 0]
+
+        return outcome
+
+
+def _load_dbot_random_teacher(model, checkpoint):
+    def interpolate_pos_embed(model, checkpoint_model):
```

Comment:
Similar to with the CLIP file, can we just make this a separate function?

Reply:
sounds good, also added reference to the original files